apps/web/src/trpc/server.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  createTRPCProxyClient,
+  loggerLink,
+  unstable_httpBatchStreamLink,
+} from "@trpc/client";
+import { cookies } from "next/headers";
+
+import { type AppRouter } from "~/server/api/root";
+import { getUrl, transformer } from "./shared";
+
+export const api = createTRPCProxyClient<AppRouter>({
+  transformer,
+  links: [
+    loggerLink({
+      enabled: (op) =>
+        process.env.NODE_ENV === "development" ||
+        (op.direction === "down" && op.result instanceof Error),
+    }),
+    unstable_httpBatchStreamLink({
+      url: getUrl(),
+      headers() {
+        return {
+          cookie: cookies().toString(),
+          "x-trpc-source": "rsc",
+        };
+      },
+    }),
+  ],
+});
```

Comment:
<i class='graphite__hidden'>[Re: lines 1 to 30]<br/></i>

move to server<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/AbhiShake1/titeet/1?utm_source=unchanged-line-comment">Graphite</a>.</span>